### PR TITLE
fast-double-parser: add version 0.5.0

### PR DESCRIPTION
**Pull request by buresu**
Specify library name and version:  fast-double-parser/0.5.0

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/fast-double-parser/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class FastDoubleParserConan(ConanFile):
+    name = "fast-double-parser"
+    description = "Fast function to parse strings into double (binary64) floating-point values, enforces the RFC 7159 (JSON standard) grammar: 4x faster than strtod"
+    topics = ("numerical", "header-only")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/lemire/fast_double_parser"
+    license = "Apache-2.0"
+
+    settings = "os", "compiler"
```
Comment (uilianries) [inline recipes/fast-double-parser/all/conanfile.py:15]:
```suggestion
    settings = "os", "compiler", "build_type", "arch"
```

Keep full settings by default. You can manage the package ID

Comment (buresu) [reply recipes/fast-double-parser/all/conanfile.py]: Thank you for reviewing! I fixed it.

File: recipes/fast-double-parser/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package_multi"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```
Comment (uilianries) [inline recipes/fast-double-parser/all/test_package/conanfile.py:15]:
```suggestion
        if not tools.cross_building(self):
```

self.settings is outdated, use only self, this new signature is better better compatible withe build context (dual profile)

Comment (buresu) [reply recipes/fast-double-parser/all/test_package/conanfile.py]: Thank you! I fixed it.

File: recipes/fast-double-parser/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import os
+from conans import ConanFile, tools
+
+required_conan_version = ">=1.43.0"
+
+
+class FastDoubleParserConan(ConanFile):
+    name = "fast-double-parser"
```
Comment (uilianries) [inline recipes/fast-double-parser/all/conanfile.py:8]:
```suggestion
    name = "fast_double_parser"
```

Please, follow the upstream's name. You also need to rename the folder name.

**Review by uilianries — APPROVED** [review]
LGTM, thank you for your contribution!

File: recipes/fast-double-parser/all/conanfile.py (hunk 3 above)
Comment (buresu) [reply recipes/fast-double-parser/all/conanfile.py]: I fixed it. Thank you a lot!

**Review by jgsogo — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/fast_double_parser/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
```
Comment (SpaceIm) [inline recipes/fast_double_parser/all/test_package/CMakeLists.txt:5]: should be `conan_basic_setup(TARGETS)`

File: recipes/fast_double_parser/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+find_package(fast_double_parser REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
```
Comment (SpaceIm) [inline recipes/fast_double_parser/all/test_package/CMakeLists.txt:9]: you don't link `fast_double_parser::fast_double_parser`

File: recipes/fast_double_parser/all/test_package/CMakeLists.txt (hunk 4 above)
Comment (buresu) [reply recipes/fast_double_parser/all/test_package/CMakeLists.txt:5]: Thank you for reviewing! I will fix them.
This MR has been merged, should I open another MR? or is it okay to overwrite push?

Comment (SpaceIm) [reply recipes/fast_double_parser/all/test_package/CMakeLists.txt:5]: you have to open another PR

Comment (buresu) [reply recipes/fast_double_parser/all/test_package/CMakeLists.txt:5]: Thank you. I will send another PR.

**Pull request merged**